sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClient.java

```diff
@@ -0,0 +1,1442 @@
+            case MIXED:
+                sentimentScores[0] = sentimentScore.getNegative();
+                sentimentScores[1] = sentimentScore.getNeutral();
+                sentimentScores[2] = sentimentScore.getPositive();
+                break;
+            default:
+                break;
+        }
+        return sentimentScores;
+    }
+
+    private TextSentimentClass convertToTextSentimentClass(final DocumentSentimentValue sentimentValue) {
+        switch (sentimentValue) {
+            case POSITIVE:
+                return TextSentimentClass.POSITIVE;
+            case NEUTRAL:
+                return TextSentimentClass.NEUTRAL;
+            case NEGATIVE:
+                return TextSentimentClass.NEGATIVE;
+            case MIXED:
+                return TextSentimentClass.MIXED;
+            default:
+                throw logger.logExceptionAsWarning(
+                    new RuntimeException(String.format("'%s' is not valid text sentiment.")));
+        }
+    }
+
+    private TextSentimentClass convertToTextSentimentClass(final SentenceSentimentValue sentimentValue) {
+        switch (sentimentValue) {
+            case POSITIVE:
+                return TextSentimentClass.POSITIVE;
+            case NEUTRAL:
+                return TextSentimentClass.NEUTRAL;
+            case NEGATIVE:
+                return TextSentimentClass.NEGATIVE;
+            default:
+                throw logger.logExceptionAsWarning(
+                    new RuntimeException(String.format("'%s' is not valid text sentiment.")));
+        }
+    }
+
+    private AnalyzeSentimentResult convertToErrorTextSentimentResult(final DocumentError documentError) {
+        final com.azure.ai.textanalytics.models.TextAnalyticsError error = convertToError(documentError.getError());
+        return new AnalyzeSentimentResult(documentError.getId(), null, error, null,
+            null);
+    }
+
+    /**
+     * Helper method to convert the service response of {@link LanguageResult} to {@link DocumentResultCollection}.
+     *
+     * @param languageResult the {@link LanguageResult} returned by the service.
+     *
+     * @return the {@link DocumentResultCollection} of {@link DetectLanguageResult} to be returned by the SDK.
+     */
+    private DocumentResultCollection<DetectLanguageResult> toDocumentResultCollection(
+        final LanguageResult languageResult) {
+        return new DocumentResultCollection<>(getDocumentLanguages(languageResult), languageResult.getModelVersion(),
+            languageResult.getStatistics() == null ? null : mapBatchStatistics(languageResult.getStatistics()));
+    }
+
+    /**
+     * Helper method to get a combined list of error documents and valid documents.
+     *
+     * @param languageResult the {@link LanguageResult} containing both the error and document list.
+     *
+     * @return the combined error and document list.
+     */
+    private List<DetectLanguageResult> getDocumentLanguages(final LanguageResult lang
```

Comment:
If these are already lists, we can do an addRange and return an existing list instead of having to traverse through both and return the result.

Reply:
i think one declared list should be sufficient?

Reply:
I mean, this entire method can be re-written to not have to initialise two arrays and then stream them. Suggestion:
```java
    private List<DetectLanguageResult> getDocumentLanguages(final LanguageResult languageResult) {
        final List<DetectLanguageResult> results = new ArrayList<>();
        for (DocumentLanguage documentLanguage : languageResult.getDocuments()) {
            results.add(new DetectLanguageResult(
                documentLanguage.getId(),
                documentLanguage.getStatistics() == null
                    ? null
                    : convertToTextDocumentStatistics(documentLanguage.getStatistics()),
                null,
                setPrimaryLanguage(documentLanguage.getDetectedLanguages()),
                convertToDetectLanguages(documentLanguage.getDetectedLanguages())));
        }

        for (DocumentError documentError : languageResult.getErrors()) {
            com.azure.ai.textanalytics.models.TextAnalyticsError error = convertToError(documentError.getError());
            results.add(new DetectLanguageResult(documentError.getId(), null, error, null,
                null));
        }

        return results;
    }
```

Reply:
updated this for all getEntity** methods.